solver/llbsolver/history.go

```diff
@@ -526,19 +526,29 @@ func (h *HistoryQueue) update(ctx context.Context, rec controlapi.BuildHistoryRe
 			return err
 		}
 		if rec.Result != nil {
-			if err := h.addResource(ctx, l, rec.Result.Result, true); err != nil {
+			if err := h.addResource(ctx, l, rec.Result.ResultDeprecated, true); err != nil {
```

Comment:
I think these changes will likely need to be tested as well?

Reply:
Added tests for the history api.